http-client/src/test/java/io/airlift/http/client/jetty/TestAsyncJettyHttpClient.java

```diff
@@ -52,4 +53,14 @@ public <T, E extends Exception> T executeRequest(HttpClientConfig config, Reques
             return executeAsync(client, request, responseHandler);
         }
     }
+
+    @Test(enabled = false)
+    protected void testPutMethodWithStreamingBodyGenerator(boolean largeContent)
+            throws Exception
+    {
+        // don't test with async clients as they buffer responses and the LARGE content is too big
```

Comment:
I need to think about it. It might be complex. Give me some time.

Reply:
tbh - the easiest thing would be to wrap the sync exec call in a virtual thread. We could do that internally when streaming is used. wdyt?

Reply:
I don't think there is a point to making the async client support streaming. You use async because you want to submit requests and get notified when they are complete, without blocking the current thread. To do streaming, you either need a completely different async API, or you block the thread to stream the results. If you're blocking the thread, then you might as well use the sync API. This is why the async API is for size limited request/response that can be fully buffered.